src/FsCheck/Gen.fs

```diff
@@ -313,6 +313,24 @@ module Gen =
     [<CompiledName("ListOf")>]
     let listOfLength n arb = sequence [ for _ in 1..n -> arb ]
 
+    ///Generates a random permutation of the given sequence using the Fisher-Yates shuffle algorithm.
+    //[category: Creating generators]
+    [<CompiledName("Shuffle")>]
+    let shuffle xs =
+        let xs = xs |> Seq.toArray
+        let swap (arr : _ array) i j =
+            let v = arr.[j]
+            arr.[j] <- arr.[i]
+            arr.[i] <- v
+        gen {
+            let copy = Array.copy xs
+            let maxI = copy.Length - 1
+            let! indexes = [ for i in 0..maxI - 1 -> choose (i, maxI) ]
```

Comment:
Wasn't this the off-by-one-bug we previously discussed? 

(Sorry to ask instead of investigating myself, but I'm on a mobile phone with only intermittent connection most of the day today, so it's a bit unwieldy to look up and experiment with branches and so on.)